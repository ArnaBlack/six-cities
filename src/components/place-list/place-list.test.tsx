import * as React from 'react';
import * as renderer from 'react-test-renderer';
import PlaceList from './place-list';

const mock = {
  offers: [
    {
      bedrooms: 1,
      city: {
        name: `Brussels`,
        location: {
          latitude: 50.846557,
          longitude: 4.351697,
          zoom: 13,
        },
      },
      description: `Description`,
      goods: [`Washer`, `Towels`],
      host: {
        avatarUrl: `img/avatar-angelina.jpg`,
        id: 25,
        isPro: true,
        name: `Name`,
      },
      id: 1,
      images: [`https://es31-server.appspot.com/six-cities/static/hotel/19.jpg`],
      isFavorite: true,
      isPremium: false,
      location: {
        latitude: 50.846557,
        longitude: 4.351697,
        zoom: 13,
      },
      maxAdults: 3,
      previewImage: `https://es31-server.appspot.com/six-cities/static/hotel/4.jpg`,
      price: 117,
      rating: 3.4,
      title: `The house among olive`,
      type: `room`,
    },
  ],
};

it(`PlaceList correctly renders`, () => {
  const {offers} = mock;
  const clickHandler = jest.fn();
  const tree = renderer
    .create(<PlaceList
      offers={offers}
      onSelectOffer={clickHandler}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
