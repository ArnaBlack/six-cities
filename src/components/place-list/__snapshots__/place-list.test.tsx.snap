// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceList correctly renders 1`] = `
<div
  className="cities__places-list places__list tabs__content"
>
  <PlaceCard
    key="1"
    offer={
      Object {
        "bedrooms": 1,
        "city": Object {
          "location": Object {
            "latitude": 50.846557,
            "longitude": 4.351697,
            "zoom": 13,
          },
          "name": "Brussels",
        },
        "description": "Description",
        "goods": Array [
          "Washer",
          "Towels",
        ],
        "host": Object {
          "avatarUrl": "img/avatar-angelina.jpg",
          "id": 25,
          "isPro": true,
          "name": "Name",
        },
        "id": 1,
        "images": Array [
          "https://es31-server.appspot.com/six-cities/static/hotel/19.jpg",
        ],
        "isFavorite": true,
        "isPremium": false,
        "location": Object {
          "latitude": 50.846557,
          "longitude": 4.351697,
          "zoom": 13,
        },
        "maxAdults": 3,
        "previewImage": "https://es31-server.appspot.com/six-cities/static/hotel/4.jpg",
        "price": 117,
        "rating": 3.4,
        "title": "The house among olive",
        "type": "room",
      }
    }
    onImageClick={[MockFunction]}
  />
</div>
`;
